src/api/src/policy_engine/opa/mod.rs

```
@@ -0,0 +1,153 @@
+            if !policy_path.as_path().parent().unwrap().exists() {
+                std::fs::create_dir_all(policy_path.parent().unwrap())?;
+            }
+            let policy = std::include_str!("default_policy.rego").to_string();
+            fs::write(&policy_path, policy)?;
+        }
+
+        Ok(Self { policy_path })
+    }
+}
+
+#[async_trait]
+impl PolicyEngineInterface for Opa {
+    async fn evaluate(
+        &self,
+        resource_path: String,
+        input_claims: String,
+    ) -> Result<(bool, String)> {
+        let policy = tokio::fs::read_to_string(
+            self.policy_path
+                .to_str()
+                .ok_or_else(|| anyhow!("Miss Policy Path"))?,
+        )
+        .await
+        .map_err(|e| anyhow!("Read OPA policy file failed: {:?}", e))?;
+
+        let policy_go = GoString {
+            p: policy.as_ptr() as *const c_char,
+            n: policy.len() as isize,
+        };
+
+        let resource_path_input = serde_json::json!({ "resource-path": resource_path }).to_string();
+
+        let resource_path_go = GoString {
+            p: resource_path_input.as_ptr() as *const c_char,
+            n: resource_path_input.len() as isize,
+        };
+
+        let input_go = GoString {
+            p: input_claims.as_ptr() as *const c_char,
+            n: input_claims.len() as isize,
+        };
+
+        // Call the function exported by cgo and process the returned decision
+        let decision_buf: *mut c_char =
+            unsafe { evaluateGo(policy_go, resource_path_go, input_go) };
+        let decision_str: &CStr = unsafe { CStr::from_ptr(decision_buf) };
+        let res = decision_str.to_str()?.to_string();
+        log::debug!("Evaluated: {}", res);
+        if res.starts_with("Error::") {
+            return Err(anyhow!(res));
+        }
+
+        let res_kv: Value = serde_json::from_str(&res)?;
+
+        Ok((res_kv["allow"].as_bool().unwrap_or(false), res))
```

Shouldn't we return an Error if `allow` isn't a bool value?